src/components/templates/PartyMap/components/Map/PartyMapRoomOverlay.tsx

```diff
@@ -14,104 +17,205 @@ import RoomAttendance from "../RoomAttendance";
+  // const currentRoomUrl = getRoomUrl(room.url);
 
-  const toggleRoom = useCallback(() => {
+  // TODO: do we still need this?
+  // const [roomClicked, setRoomClicked] = useState<string | undefined>(undefined);
+  // TODO: do we need the whole room object? Can we just roll this logic into whatever is setting roomClicked, then rename above to isRoomClicked?
+  // const isThisRoomClicked = roomClicked === room.title;
+
+  // TODO: I think this may be intended to toggle 'clicked' for a room, do we need to separately keep track of 'clicked'? Or is it the same as selecting the room?
+  // TODO: what is a better semantic name for this?
+  const onClick1 = useCallback(() => {
+    // if (!IS_BURN) {
     selectRoom(room);
+    // } else {
+    // setRoomClicked((prevRoomClicked) =>
+    //   prevRoomClicked === room.title ? undefined : room.title
+    // );
+    // }
   }, [room, selectRoom]);
 
-  const [isRoomHovered, setRoomHovered] = useState<
-    PartyMapRoomData | undefined
-  >(undefined);
+  // TODO: this was used in place of onClick1 in PartyMap Map before I merged them, is it correct? Or is the onClick1 definition above?
+  // const toggleRoom = useCallback(() => {
+  //   selectRoom(room);
+  // }, [room, selectRoom]);
+
+  const [roomHovered, setRoomHovered] = useState<PartyMapRoomData | undefined>(
+    undefined
+  );
+
+  const isRoomHovered = isTruthy(roomHovered);
 
   const dispatch = useDispatch();
 
-  const roomHovered = useCallback(() => {
-    dispatch(retainAttendance(true));
+  const handleRoomHovered = useCallback(() => {
     setRoomHovered(room);
+    dispatch(retainAttendance(true));
   }, [dispatch, room]);
 
-  const roomUnhovered = useCallback(() => {
-    dispatch(retainAttendance(false));
+  const handleRoomUnhovered = useCallback(() => {
     setRoomHovered(undefined);
+    dispatch(retainAttendance(false));
   }, [dispatch]);
 
-  const left = room.x_percent;
-  const top = room.y_percent;
-  const width = room.width_percent;
-  const height = room.height_percent;
+  // TODO: this is being spread here because the below code modifies it with .push()/etc
+  // const rooms = [...venue.rooms];
+
+  // TODO: what is the intent/purpose of this?
+  // TODO: how do we implement this nicely, and where?
+  // TODO: reinstate this functionality
+  // if (roomHovered) {
+  //   const idx = rooms.findIndex((room) => room.title === roomHovered.title);
+  //   if (idx !== -1) {
+  //     const chosenRoom = rooms.splice(idx, 1);
+  //     rooms.push(chosenRoom[0]);
+  //   }
+  // }
+
+  // TODO: what is a better semantic name for this?
+  // const onJoinRoom: React.MouseEventHandler = useCallback(
+  //   (e) => {
+  //     e.stopPropagation();
+  //
+  //     if (isExternalUrl(room.url)) {
+  //       openRoomUrl(room.url);
+  //     } else {
+  //       window.location.href = getRoomUrl(room.url);
+  //     }
+  //
+  //     enterCampRoom(room);
+  //   },
+  //   [enterCampRoom, room]
+  // );
```

Comment:
- [ ] [TODO] what is a better semantic name for this?
- [ ] [TODO] do we need to reinstate this? Is it handled somewhere already?

Reply:
`enterCampRoom` essentially calls `enterRoom` which calls `updateLocationData` which relates to where firestore thinks the user is located (which is used for counting)

We can probably rename `enterRoom` something more useful, and/or roll the 'open link' stuff into it as well.

Reply:
resolved in https://github.com/sparkletown/sparkle/pull/840/commits/95c2222ce28d237fb33b44dd69747151fd89e865 + https://github.com/sparkletown/sparkle/pull/840/commits/4b6a1669636861e45d8f7344d9ba82a8437b29c8